txscript/data/script_tests.json

```diff
@@ -1365,87 +1459,193 @@
+["-1 -1", "NUMEQUAL", "NONE", "OK"],
+["1", "0x02 0x0100 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Numerically equal but rejected by minimal data encoding requirements"],
+["11", "0x4c 0x03 0x0b0000 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0", "0x01 0x80 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Zero numerically equals negative zero but rejected by minimal data encoding requirements"],
+["0", "0x02 0x0080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Zero numerically equals negative zero with padding but rejected by minimal data encoding requirements"],
+["0x03 0x000080", "0x04 0x00000080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0x03 0x100080", "0x04 0x10000080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0x03 0x100000", "0x04 0x10000000 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["1 2147483648 NUMEQUAL", "NOT", "NONE", "ERR_OUT_OF_RANGE", "NUMEQUAL must fail with stack item >4 bytes"],
+
+["Equivalency of different numeric encodings"],
+["0x02 0x8000", "128 NUMEQUAL", "NONE", "OK", "0x8000 equals 128"],
+["0x01 0x00", "0 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x00 numequals 0 but rejected by minimal data encoding requirements"],
+["0x01 0x80", "0 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x80 (negative zero) numequals 0 but rejected by minimal data encoding requirements"],
+["0x02 0x0080", "0 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x0080 numequals 0 but rejected by minimal data encoding requirements"],
+["0x02 0x0500", "5 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x0500 numequals 5 but rejected by minimal data encoding requirements"],
+["0x03 0xff7f80", "0x02 0xffff NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xff7f80 numequals 0xfffff but rejected by minimal data encoding requirements"],
+["0x03 0xff7f00", "0x02 0xff7f NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xff7f00 numequals 0xff7f but rejected by minimal data encoding requirements"],
+["0x04 0xffff7f80", "0x03 0xffffff NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xffff7f80 numequals 0xffffff but rejected by minimal data encoding requirements"],
+["0x04 0xffff7f00", "0x03 0xffff7f NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xffff7f00 numequals 0xffff7f but rejected by minimal data encoding requirements"],
+
+["NUMEQUALVERIFY test coverage"],
+["NOP NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items and NOP must not be treated as one"],
+["NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items, stack has no items"],
+["0 NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items, stack has one item"],
+["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK", "NUMEQUALVERIFY must compare the first numeric stack item to the second numeric stack item, check if they are equal and verify the result"],
+["-1 -1", "NUMEQUALVERIFY 1", "NONE", "OK"],
+["1 2147483648", "NUMEQUALVERIFY 1", "NONE", "ERR_OUT_OF_RANGE", "NUMEQUALVERIFY must fail with stack item >4 bytes"],
```

Comment:
Same here regarding needing another test for the other operand being >4 bytes and comment.